diff --git a/kong/conf_loader.lua b/kong/conf_loader.lua
index 675ad0c..90742e5 100644
--- a/kong/conf_loader.lua
+++ b/kong/conf_loader.lua
@@ -98,6 +98,9 @@ local CONF_INFERENCES = {
   ssl = {typ = "boolean"},
   client_ssl = {typ = "boolean"},
   admin_ssl = {typ = "boolean"},
+  admin_basic_auth = {typ = "boolean"},
+  admin_acl = {typ = "string"},
+  admin_basic_auth_file = {typ = "string"},

   proxy_access_log = {typ = "string"},
   proxy_error_log = {typ = "string"},
diff --git a/kong/templates/kong_defaults.lua b/kong/templates/kong_defaults.lua
index 6f79bb4..5a3b30a 100644
--- a/kong/templates/kong_defaults.lua
+++ b/kong/templates/kong_defaults.lua
@@ -27,6 +27,9 @@ ssl_ciphers = ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-EC
 admin_ssl = on
 admin_ssl_cert = NONE
 admin_ssl_cert_key = NONE
+admin_basic_auth = off
+admin_acl = 127.0.0.1/32
+admin_basic_auth_file = /etc/kong/kong.admin.htpasswd
 upstream_keepalive = 60
 server_tokens = on
 latency_tokens = on
diff --git a/kong/templates/nginx_kong.lua b/kong/templates/nginx_kong.lua
index 3c03859..e49c6bc 100644
--- a/kong/templates/nginx_kong.lua
+++ b/kong/templates/nginx_kong.lua
@@ -174,6 +174,13 @@ server {
 > end

     location / {
+> if admin_basic_auth then
+        satisfy              any;
+        allow                ${{ADMIN_ACL}};
+        deny                 all;
+        auth_basic           "kong admin";
+        auth_basic_user_file ${{ADMIN_BASIC_AUTH_FILE}};
+> end
         default_type application/json;
         content_by_lua_block {
             ngx.header['Access-Control-Allow-Origin'] = '*'
