--- templates/nginx_kong.lua.orig 2017-05-26 05:40:12.367382249 +0000
+++ templates/nginx_kong.lua  2017-05-26 05:40:02.623358816 +0000
@@ -155,6 +155,9 @@
 > end
 
     location / {
+        satisfy  any;
+        allow 127.0.0.1/32;
+        deny   all;
         auth_basic           "${{ADMIN_AUTH_BASIC}}";
         auth_basic_user_file ${{ADMIN_AUTH_FILE}};
         default_type application/json;

