--- templates/nginx_kong.lua.orig 2017-05-26 05:24:47.373157396 +0000
+++ templates/nginx_kong.lua  2017-05-26 05:25:50.989310428 +0000
@@ -155,6 +155,8 @@
 > end
 
     location / {
+        auth_basic           "${{ADMIN_AUTH_BASIC}}";
+        auth_basic_user_file ${{ADMIN_AUTH_FILE}};
         default_type application/json;
         content_by_lua_block {
             ngx.header['Access-Control-Allow-Origin'] = '*'
